### ISSUE TITLE
New release of jsonschema breaks altair

### ISSUE THREAD

**Issue by freddyaboulton** [altair-viz/altair#3105]
The new jsonschema release breaks altair.

Reproducer: https://colab.research.google.com/drive/11sSGtDueOh8Ig3ZKFvKhRqsF4IWnkAmN?usp=sharing

```bash
pip install altair jsonschema --upgrade
```

```python
import altair as alt
from vega_datasets import data

source = data.cars()

# Brush for selection
brush = alt.selection_interval()

# Scatter Plot
points = alt.Chart(source).mark_point().encode(
    x='Horsepower:Q',
    y='Miles_per_Gallon:Q',
    color=alt.condition(brush, alt.value('steelblue'), alt.value('grey'))
).add_params(brush)

# Base chart for data tables
ranked_text = alt.Chart(source).mark_text(align='right').encode(
    y=alt.Y('row_number:O').axis(None)
).transform_filter(
    brush
).transform_window(
    row_number='row_number()'
).transform_filter(
    alt.datum.row_number < 15
)

# Data Tables
horsepower = ranked_text.encode(text='Horsepower:N').properties(
    title=alt.Title(text='Horsepower', align='right')
)
mpg = ranked_text.encode(text='Miles_per_Gallon:N').properties(
    title=alt.Title(text='MPG', align='right')
)
origin = ranked_text.encode(text='Origin:N').properties(
    title=alt.Title(text='Origin', align='right')
)
text = alt.hconcat(horsepower, mpg, origin) # Combine data tables

# Build chart
alt.hconcat(
    points,
    text
).resolve_legend(
    color="independent"
).configure_view(
    stroke=None
)
```

Error `_RefResolutionError: 'bytes' object has no attribute 'timeout'`

Seems related to #3097

---

Please follow these steps to make it more efficient to solve your issue:

- [ ] Since Altair is a Python wrapper around the Vega-Lite visualization grammar, [most bugs should be reported directly to Vega-Lite](https://github.com/vega/vega-lite/issues). You can click the Action Button of your Altair chart and "Open in Vega Editor" to create a reproducible Vega-Lite example and see if you get the same error in the Vega Editor.
- [ ] Search for duplicate issues.
- [ ] Use the latest version of Altair.
- [ ] Describe how to reproduce the bug and include the full code and data to reproduce it, ideally using a sample data set from `vega_datasets`.

**Comment by binste** [comment]
Thank you for reporting this. This was a regression on the jsonschema side which they now fixed in a new version 4.18.2. See https://github.com/python-jsonschema/jsonschema/issues/1124 for reference. So upgrading to 4.18.2 should solve it 🙂

**Comment by freddyaboulton** [comment]
I appreciate the quick fix @binste ! Thanks
